### ISSUE TITLE
[java] InlineImmediateVariable S1488: Local variables should not be declared and then immediately used

### ISSUE THREAD

**Issue by Pankraz76** [pmd/pmd#5771]
- https://github.com/pmd/pmd/issues/5770
- https://github.com/apache/maven/pull/2404
- https://github.com/apache/maven/pull/2404/files#r2113612293
- https://github.com/apache/maven/pull/2417/files#r2119215587
- https://checkstyle.sourceforge.io/checks/coding/variabledeclarationusagedistance.html
- https://en.wikipedia.org/wiki/Coupling_(computer_programming)

**InlineImmediateVariable** this seems profound.
**InlineOneTimeVariable** this sounds not well.

**Proposed Category:** One of [Best Practices]

**Description:**

Inline variables when they are immediately **and only one time** used.

**Code Sample:** This should include code, that should be flagged by the rule. If possible, the "correct" code
according to this new rule should also be demonstrated.

```java
    private synchronized void ensureInitialized() {
        if (!initialized) {
            Map<String, String> initial = getter.get();
            orderedProps.putAll(initial);
            initialized = true;
        }
    }
```

**type does not matter at all, its random impl. detail.**

```java
    private synchronized void ensureInitialized() {
        if (!initialized) {
            orderedProps.putAll(getter.get());
            initialized = true;
        }
    }
```

**Possible Properties:**

*   Should this rule be customizable via properties?

**Comment by zbynek** [comment]
If you fix this violation repeatedly, you might end up compressing a whole method into a single statement, e.g.
```
Foo foo=getA().getB().getC();
Bar bar=getF().getG().getH();
Baz baz=getX().getY().getZ();
foo.check(bar, baz);
```
would become
```
getA().getB().getC().check(getF().getG().getH(), getX().getY().getZ());
```

IMHO that just makes it harder to read and harder to put breakpoints to the right places.

The more restrictive version of this rule proposed in https://github.com/pmd/pmd/issues/5770 seems more useful.

**Comment by Pankraz76** [comment]
```
getA().getB().getC().
    check(
        getF().getG().getH(), 
        getX().getY().getZ());
```
yes as we all rely on formation its functional.
Thanks for the good example.

**Comment by Pankraz76** [comment]
Kindly consider this example for automation, spotless and other tools can do the same:
- https://github.com/junit-team/junit-framework/pull/4708

**Comment by Pankraz76** [comment]
of course this rule is just some example, not really ment to be implemented, as this would end up in pure functional code, which is not the standard anyone wants or is able to reach.

**Comment by adangel** [comment]
@zbynek thanks for the example.

@Pankraz76 Ok, I think we can agree to close this issue then.

**Comment by Pankraz76** [comment]
assumingi its an good extention of the initial rule.

If there is 2 (max 3) vars, there is a good reason to inline as shown.

Proving the star or result.

- https://github.com/gradle/gradle/pull/34980/files#r2347566777
